### ISSUE TITLE
CocoaPods is not using Conditional Variable Assignment

### ISSUE THREAD

**Issue by ppeelen** [CocoaPods/CocoaPods#7133]
**Cocoapods version**: 1.3.1</br>
**Xcode version**: 9.0 (9A235)

Cocoapods is generating a lot of xcconfig files which creates overhead when having custom xcconfig files in project.

I expected cocoapods to generate one `xcconfig` file per target using conditional variable assignment in each file for each scheme. Instead, for each scheme and each target an `xcconfig` file was created. Many of which (if not all) with the same settings.

This creates a big problem for projects with multiple targets and multiple schemes. In our case we have 5 schemes (`Debug`, `Release`, `Profile`, `AdHoc` and `AdHoc Dev`) and about 9 targets. 
When we would like to create specific config files for some of our targets, it means that we need to create 6 `xcconfig` files for each target.

Eg.
`shared.xcconfig` which sets settings like version number ect.
`App.xcconfig` which sets the different apps config setting.
`App-{configuration}.xcconfig` which includes the `App.xcconfig` and Pods configuration only.
This creates a lot of overhead.

## Current:

**Shared.xcconfig**:

```
BUNDLE_VERSION = 4.4.0
BUILD_NUMBER = 1
```

**App1.xcconfig**:

```
#include "Shared.xcconfig"
BUNDLE_IDENIFIER = "net.myapp.app1"
```

**App1-Debug.xcconfig**:

```
#include "App1.xcconfig"
#include "Pods/Target Support Files/Product1/Pods-App1.Debug.xcconfig"
```

## Requested

Instead I would like cocapods to create one `Pods/Target Support Files/Product1/Pods-App1.xcconfig` and have the following example content (of course, the below is shortened and values are incorrect):

```
ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES[config=Debug] = YES
ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES[config=Release] = NO

FRAMEWORK_SEARCH_PATHS[config=Debug] = $(inherited) ""
FRAMEWORK_SEARCH_PATHS[config=Release] = $(inherited) "Something different"

PODS_CONFIGURATION_BUILD_DIR[config=Debug] = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)

PODS_PODFILE_DIR_PATH[config=Debug] = ${SRCROOT}/.

PODS_ROOT[config=Debug] = ${SRCROOT}/Pods
```

Hence the addition of conditional variable assignment using `[config=Debug]` and `[config=Release]`. More information [here](https://pewpewthespells.com/blog/xcconfig_guide.html#ConditionalVariableAssignment).

That way I can assign `App1.xcconfig` in Xcode to `App1` under `Configuration Debug` with the contents of `App1.xcconfig` being the following:

**App1.xcconfig**:

```
#include "Shared.xcconfig"
#include "Pods/Target Support Files/Product1/Pods-App1.xcconfig"

BUNDLE_IDENIFIER = "net.myapp.app1"
```

I hope my feature request was propperly written according to your guidelines. I also believe that this is something many users might have use of. Especially those with large projects.

**Comment by dnkoutso** [comment]
@ppeelen thanks for this enhancement request. I personally need to think it through myself but this would be a larger change in how CocoaPods works.

Generally the xcconfig file generation has been the slowest part in a large project and this would help with performance but there are other things to consider before implementing this.

**Comment by ppeelen** [comment]
Of course, I understand that this is not an easy item to tackle. A hint would be to have cocoapods to be backwards compatible using a variable in the Podsfile. E.g. `use_cva` or `use_seperate_xcconfig`.

**Comment by hovox** [comment]
hey guys,

Is there any work done on this issue ? We also have same exact issue.
